### PR TITLE
[TO_REVIEW] Fix Circular validation for NO_DA_TARGET_ONLY

### PR DESCRIPTION
**Pull request by YanisLalou**
Before when using the Circular validation while using the solver NO_DA_TARGET_ONLY:

```
Traceback (most recent call last):
  File "/Users/yanislalou/Documents/CMAP/skada_test_venv/lib/python3.9/site-packages/sklearn/metrics/_scorer.py", line 141, in __call__
    score = scorer(estimator, *args, **routed_params.get(name).score)
  File "/Users/yanislalou/Documents/CMAP/skada_test_venv/lib/python3.9/site-packages/skada/metrics.py", line 43, in __call__
    return self._score(estimator, X, y, sample_domain=sample_domain, **params)
  File "/Users/yanislalou/Documents/CMAP/skada_test_venv/lib/python3.9/site-packages/skada/metrics.py", line 586, in _score
    backward_estimator.fit(X, backward_y, sample_domain=backward_sample_domain)
  File "/Users/yanislalou/Documents/CMAP/skada_test_venv/lib/python3.9/site-packages/sklearn/base.py", line 1351, in wrapper
    return fit_method(estimator, *args, **kwargs)
  File "/Users/yanislalou/Documents/CMAP/skada_test_venv/lib/python3.9/site-packages/sklearn/pipeline.py", line 475, in fit
    self._final_estimator.fit(Xt, y, **last_step_params["fit"])
  File "/Users/yanislalou/Documents/CMAP/skada_test_venv/lib/python3.9/site-packages/skada/base.py", line 543, in fit
    return self._pre_filter('fit', X_container, y=y, **params)
  File "/Users/yanislalou/Documents/CMAP/skada_test_venv/lib/python3.9/site-packages/skada/base.py", line 539, in _pre_filter
    return getattr(super(), method_name)(X_container.merge_in((X_input, y, params)), y, **params)
  File "/Users/yanislalou/Documents/CMAP/skada_test_venv/lib/python3.9/site-packages/skada/base.py", line 390, in fit
    self._fit('fit', X_container, y, **params)
  File "/Users/yanislalou/Documents/CMAP/skada_test_venv/lib/python3.9/site-packages/skada/base.py", line 413, in _fit
    output = getattr(estimator, routing_method)(X, y, **routed_params)
  File "/Users/yanislalou/Documents/CMAP/skada-bench/benchmark_utils/base_solver.py", line 47, in fit
    self.estimator_.fit(X, y, sample_weight=sample_weight, **fit_params)
  File "/Users/yanislalou/Documents/CMAP/skada_test_venv/lib/python3.9/site-packages/sklearn/base.py", line 1351, in wrapper
    return fit_method(estimator, *args, **kwargs)
  File "/Users/yanislalou/Documents/CMAP/skada_test_venv/lib/python3.9/site-packages/sklearn/linear_model/_logistic.py", line 1201, in fit
    X, y = self._validate_data(
  File "/Users/yanislalou/Documents/CMAP/skada_test_venv/lib/python3.9/site-packages/sklearn/base.py", line 650, in _validate_data
    X, y = check_X_y(X, y, **check_params)
  File "/Users/yanislalou/Documents/CMAP/skada_test_venv/lib/python3.9/site-packages/sklearn/utils/validation.py", line 1192, in check_X_y
    X = check_array(
  File "/Users/yanislalou/Documents/CMAP/skada_test_venv/lib/python3.9/site-packages/sklearn/utils/validation.py", line 1026, in check_array
    raise ValueError(
ValueError: Found array with 0 sample(s) (shape=(0, 30)) while a minimum of 1 is required by LogisticRegression.
```

The problem was the target labels which were set to -1. However when using the SelectTarget selector we need to have unmasked labels.

### CODE REVIEW

File: skada/metrics.py [hunk 1]
```diff
@@ -564,25 +562,20 @@ def _score(self, estimator, X, y, sample_domain=None):
...
             backward_sample_domain = -sample_domain
 
-            backward_y = np.zeros_like(y)
-            backward_y[~source_idx] = y_pred_target
-
-            if y_type == Y_Type.DISCRETE:
-                backward_y[source_idx] = _DEFAULT_MASKED_TARGET_CLASSIFICATION_LABEL
-            else:
-                backward_y[source_idx] = _DEFAULT_MASKED_TARGET_REGRESSION_LABEL
-
```
Comment (kachayev) [inline skada/metrics.py:585]: I'm not sure that these lines need to be removed. You still need to mask the part of the input that is not considered to be targets, right? 🤔

Comment (YanisLalou) [reply skada/metrics.py:585]: Well every DA methods implemented in skada are supposed to not using the target labels, so feeding them to the methods wont change anything. And with the circular scorer, when we train on the target domain we need to unmasked labels (or it breaks).

I know that usually we mask them, to be sure that we don't have any data leak, bu here I don't see any other solutions

Comment (kachayev) [reply skada/metrics.py:585]: I'm not sure what you mean by 'not suppose to use target labels'. The estimator that doesn't request `sample_domain` gets input without separation for source/target (as it cannot operate with it), only masked labels are stripped away.

**Comment by kachayev** [comment]
Ready for re-review? @YanisLalou

**Review by kachayev — APPROVED** [review]

**Pull request merged**